internal/sharedtest/hooks.go

```diff
@@ -0,0 +1,169 @@
+package sharedtest
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"golang.org/x/exp/slices"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/launchdarkly/go-sdk-common/v3/ldreason"
+	"github.com/launchdarkly/go-server-sdk/v7/ldhooks"
+)
+
+// HookStage is the stage of a hook being executed.
+type HookStage string
+
+const (
+	// HookStageBeforeEvaluation is the stage executed before evaluation.
+	HookStageBeforeEvaluation HookStage = "before"
+	// HookStageAfterEvaluation is the stage executed after evaluation.
+	HookStageAfterEvaluation = "after"
+)
```

Comment:
I always found it ambiguous if the second const is actually a raw string or the alias. (I forget which.)

So I do:
```suggestion
const (
	// HookStageBeforeEvaluation is the stage executed before evaluation.
	HookStageBeforeEvaluation = HookStage("before")
	// HookStageAfterEvaluation is the stage executed after evaluation.
	HookStageAfterEvaluation = HookStage("after")
)
```